code/__HELPERS/view.dm

```diff
@@ -20,8 +20,21 @@
 	view_info[2] *= world.icon_size
 	return view_info
 
-/proc/in_view_range(mob/user, atom/A)
-	var/list/view_range = getviewsize(user.client.view)
-	var/turf/source = get_turf(user)
-	var/turf/target = get_turf(A)
-	return ISINRANGE(target.x, source.x - view_range[1], source.x + view_range[1]) && ISINRANGE(target.y, source.y - view_range[1], source.y + view_range[1])
+/**
+* Frustrated with bugs in can_see(), this instead uses viewers for a much more effective approach.
+* ### Things to note:
+* - Src/source must be a mob. `viewers()` returns mobs.
+* - Adjacent objects are always considered visible.
+* - Default distance is 7.
+*/
+
+/// Basic check to see if the src object can see the target object.
+#define CAN_I_SEE(target) ((src in viewers(7, target)) || in_range(target, src))
+
+
+/// Checks the visibility between two other objects.
+#define CAN_THEY_SEE(target, source) ((source in viewers(7, target)) || in_range(target, source))
```

Comment:
```suggestion
*/

/// The default tile-distance between two atoms for one to consider the other as visible.
#define DEFAULT_SIGHT_DISTANCE 7

/// Basic check to see if the src object can see the target object.
#define CAN_I_SEE(target) ((src in viewers(DEFAULT_SIGHT_DISTANCE, target)) || in_range(target, src))


/// Checks the visibility between two other objects.
#define CAN_THEY_SEE(target, source) ((source in viewers(DEFAULT_SIGHT_DISTANCE, target)) || in_range(target, source))
```

bit more self-documenting and easier to change in case we need to in the future.